optional_plugins/robot/avocado_robot/__init__.py

```diff
@@ -0,0 +1,131 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2017
+# Authors: Amador Pahim <apahim@redhat.com>
+
+import collections
+import subprocess
+
+from avocado.core import exceptions
+from avocado.core import loader
+from avocado.core import output
+from avocado.core import test
+from avocado.core.plugin_interfaces import CLI
+from avocado.core.runner import TestRunner
+from avocado.utils import path as utils_path
+from robot.parsing.model import TestData
+
+
+ROBOT_BINARY_NAME = 'robot'
+
+
+class RobotTest(test.SimpleTest):
+
+    """
+    Run a Robot command as a SIMPLE test.
+    """
+
+    def __init__(self,
+                 name,
+                 params=None,
+                 base_logdir=None,
+                 job=None,
+                 external_runner=None):
+        super(RobotTest, self).__init__(name, params, base_logdir, job)
+
+    @property
+    def filename(self):
+        """
+        Returns the path of the robot test suite.
+        """
+        return self.name.name.split(':')[0]
+
+    def test(self):
+        """
+        Create the Robot command and execute it.
+        """
+        try:
+            robot_binary = utils_path.find_command(ROBOT_BINARY_NAME)
+        except utils_path.CmdNotFoundError:
+            raise exceptions.TestError('"%s" binary not found.' %
+                                       ROBOT_BINARY_NAME)
+
+        suite_name, test_name = self.name.name.split(':')[1].split('.')
+        cmd = '%s --suite "%s" --test "%s" %s' % (robot_binary,
+                                                  suite_name,
+                                                  test_name,
+                                                  self.filename)
```

Comment:
Yes, actually I remember that I started with that last week and than changed so we can run as subprocesses. I'm adding some documentation on that. Thank you.